### PR TITLE
Add getEncryptionPublicKey on KeyringController

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## Description

This PR adds the `getEncryptionPublicKey` method on `KeyringController`.

This method calls the homonym `EthKeyringController` method, which returns the encryption public key of the specified address.

## Changes

- **ADDED**: `getEncryptionPublicKey` method

## References

* Fixes: #1556

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by mcmire — COMMENTED** [review]
Is there a way we can add tests for this? Perhaps we can take inspiration from the corresponding test in `eth-keyring-controller`: https://github.com/MetaMask/KeyringController/blob/12e3a2017a9f9b76e10497d93746deeefe3e190b/src/KeyringController.test.ts#L1142

File: packages/keyring-controller/src/KeyringController.ts [hunk 1]
```diff
@@ -441,6 +441,20 @@ export class KeyringController extends BaseControllerV2<
     return this.#keyring.getKeyringForAccount(account);
   }
 
+  /**
+   * Returns the encryption public key of
+   * the specified `address` if one exists.
+   *
+   * @deprecated Use of this method is discouraged as actions executed directly on
+   * keyrings are not being reflected in the KeyringController state and not
+   * persisted in the vault.
```
Comment (mikesposito) [inline packages/keyring-controller/src/KeyringController.ts:450]: We don't need to deprecate this one, as the client calling this method will not have direct access to the underlying EthKeyringController or Keyring.

Moreover, `this.#keyring.getEncryptionPublicKey(account)` does not apply any change to the controller state

File: packages/keyring-controller/src/KeyringController.ts [hunk 2]
```diff
@@ -441,6 +441,20 @@ export class KeyringController extends BaseControllerV2<
...
+   * @param account - An account address.
+   * @returns Promise resolving to encyption public key of the `account` if one exists.
```
Comment (mikesposito) [inline packages/keyring-controller/src/KeyringController.ts:452]: Nit: we can specify that this method throws an exception when there's no keyring managing that address and when the keyring does not support public key encryption:

```suggestion
   * @returns Promise resolving to encyption public key of the `account` if one exists.
   * @throws If the `account` does not exist or does not support the `getEncryptionPublicKey` method
```

**Review by mikesposito — CHANGES_REQUESTED** [review]

File: packages/keyring-controller/src/KeyringController.ts [hunk 3]
```diff
@@ -427,6 +427,17 @@ export class KeyringController extends BaseControllerV2<
...
   }
 
+  /**
+   * Get encryption public key.
+   *
+   * @param account - An account address.
+   * @throws If the `account` does not exist or does not support the `getEncryptionPublicKey` method
+   * @returns Promise resolving to encyption public key of the `account` if one exists.
+   */
+  async getEncryptionPublicKey(account: string): Promise<string> {
```
Comment (mikesposito) [inline packages/keyring-controller/src/KeyringController.ts:437]: We should also support an optional argument `opts?: Record<string, unknown>` argument to pass to `this.#keyring.getEncryptionPublicKey`

See: https://github.com/MetaMask/KeyringController/blob/main/src/KeyringController.ts#L467

**Review by mikesposito — COMMENTED** [review]
Looks good! I think we just miss an optional argument

**Review by mikesposito — APPROVED** [review]

**Pull request merged**